### ISSUE TITLE
Link: Cryptic output if no main function found

### ISSUE THREAD

**Issue by jonnyzzz** [JetBrains/kotlin-native#772]
I created a hello-world app. By mistake, I placed the `main` function into a package. 
The result was cryptic:
```
Successfully started process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/bin/java''
KtFile: kpm.kt
Undefined symbols for architecture x86_64:
  "_EntryPointSelector", referenced from:
      _Konan_start in combined5325091151054523226.o
ld: symbol(s) not found for architecture x86_64
exception: java.lang.IllegalStateException: The /Users/jonnyzzz/Work/kotlin-native/dist/dependencies/target-sysroot-1-darwin-macos/usr/bin/ld command returned non-zero exit code: 1.
        at org.jetbrains.kotlin.backend.konan.LinkStage.runTool(LinkStage.kt:316)
        at org.jetbrains.kotlin.backend.konan.LinkStage.link(LinkStage.kt:269)
        at org.jetbrains.kotlin.backend.konan.LinkStage.access$link(LinkStage.kt:187)
        at org.jetbrains.kotlin.backend.konan.LinkStage$linkStage$3.invoke(LinkStage.kt:335)
        at org.jetbrains.kotlin.backend.konan.LinkStage$linkStage$3.invoke(LinkStage.kt:187)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:129)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:108)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:128)
        at org.jetbrains.kotlin.backend.konan.LinkStage.linkStage(LinkStage.kt:334)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$5.invoke(KonanDriver.kt:111)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$5.invoke(KonanDriver.kt)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:129)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:108)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:128)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:110)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:69)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:56)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:93)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:46)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:92)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:70)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:36)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:157)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:148)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:169)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:166)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profileIf(util.kt:33)
        at org.jetbrains.kotlin.backend.konan.util.UtilKt.profile(util.kt:28)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion.main(K2Native.kt:168)
        at org.jetbrains.kotlin.cli.bc.K2NativeKt.main(K2Native.kt:174)


:src:kpm-cli:compileKonanKpm (Thread[Task worker,5,main]) completed. Took 5.779 secs.
```

Let's make more descriptive error message for the case I missing the correct entrypoint. Compiler may warn on matching entrypoints to simplify self healiing.

**Comment by ilmat192** [comment]
This issue doesn't relate to Gradle and you can observe it using the command line compiler too. But I agree, such messages should be made more informative.

**Comment by alexander-gorshenev** [comment]
Fixed by #827.
